src/WebJobs.Script/FileTraceWriter.cs

```diff
@@ -184,11 +184,9 @@ private void OnFlushLogs(object sender, ElapsedEventArgs e)
 
         internal void SetNewLogFile()
         {
-            // purge any log files over our retention count
-            // we keep MaxLogFileCount - 1 since we're moving to a new log file below
-            // and once it is written to we'll be at MaxLogFileCount files
-            var files = GetLogFiles(_logDirectory);
-            var filesToPurge = files.Skip(MaxLogFileCount - 1);
+            // purge any log files (regardless of instance ID) whose last write time was earlier
+            // than our retention policy
+            var filesToPurge = _logDirectory.GetFiles("*.log").Where(p => (DateTime.Now - p.LastWriteTime).TotalDays > LastModifiedCutoffDays);
```

Comment:
Purging across instance IDs helps us deal with the fact that there are 1000s of VMs in the Dynamic Compute stamp. We want to ensure we clean up log files in the event that a particular app never runs on a particular instance again.